### DISCUSSION TITLE
Custom Currency Symbol

### DISCUSSION THREAD

**Discussion by casesandberg** [barvian/number-flow#44] [Ideas]
Is there a way to provide a custom currency symbol? Im currently using `Intl.NumberFormat` and then `.replace('$', currencySymbol)`, however not sure how to do that with your component.

**Comment by barvian** [comment]
Hi 👋, not at the moment. Can I ask what you're replacing it with? Converting this to a discussion for voting

**Comment by rcoopr** [comment]
I'm not sure about a custom currency, but I also have a use case where instead of displaying 0 I'd like to display a `-`. It's a minor nitpick though since I can just fade out the component and fade in a static span instead

**Comment by nordowl** [comment]
A custom currency might be useful if you wanted to display crypto currencies, like BTC and ETH.

**Comment by barvian** [comment]
I'm planning some type of overrides feature to better handle this 👍. For reference, Intl.NumberFormat already handles currencies like BTC and ETH (albeit not their symbols): [Sandbox](https://codesandbox.io/p/sandbox/number-flow-react-forked-t7lgh9?workspaceId=cbc394d8-76be-4eee-b6ad-557eaf0d663f)

**Reply by nordowl** [reply]
> For reference, Intl.NumberFormat already handles currencies like BTC and ETH

Interesting. Learned something new

**Comment by barvian** [comment]
It's not quite the original request but I just pushed `prefix`/`suffix` support which should help with this.

**Comment by rnbrady** [comment]
Prefix and suffix are great solution. For those needing more flexbility you could write a small polyfill for `Intl.NumberFormat` (demo [here](https://playcode.io/2150694)). 

<img width="706" alt="Screenshot 2024-11-17 at 10 28 16" src="https://github.com/user-attachments/assets/dc2b8c29-6425-48e0-b554-b8d942d7f375">

I was doing this prior to latest release but have switched over to `prefix` which is a much cleaner solution. Thanks @barvian!
